stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -1172,15 +1167,13 @@ extension Set {
   ///   `false`.
   public static func == (lhs: Set<Element>, rhs: Set<Element>) -> Bool {
     switch (lhs._variantStorage, rhs._variantStorage) {
-    case (.native(let lhsNativeOwner), .native(let rhsNativeOwner)):
-      let lhsNative = lhsNativeOwner.nativeStorage
-      let rhsNative = rhsNativeOwner.nativeStorage
+    case (.native(let lhsNative), .native(let rhsNative)):
 
-      if lhsNativeOwner === rhsNativeOwner {
+      if lhsNative === rhsNative {
         return true
       }
 
-      if lhsNative.count != rhsNative.count {
+      if lhsNative._count != rhsNative._count {
```

Comment:
What does it mean for a field to be an implementation detail, when the containing type is entirely implementation detail? Just that the other parts that use it shouldn't directly make use of that particular implementation detail? Why aren't we just taking advantage of Swift's privacy system for this?

To be clear, all of the Storage/Buffer types, I believe, could be marked fileprivate with no impact.